pyogrio/tests/test_geopandas_io.py

```diff
@@ -479,29 +515,136 @@ def test_write_dataframe_gdalparams(tmp_path, naturalearth_lowres):
     assert test_withindex_index_filename.exists() is True
 
 
-def test_write_dataframe_geometry_type(tmp_path, naturalearth_lowres):
-    df = read_dataframe(naturalearth_lowres)
+@pytest.mark.parametrize(
+    "ext, promote_to_multi, expected_geometry_types, expected_layer_geometry_type",
+    [
+        (".fgb", None, ["MultiPolygon"], "MultiPolygon"),
+        (".fgb", True, ["MultiPolygon"], "MultiPolygon"),
+        (".fgb", False, ["MultiPolygon", "Polygon"], "Unknown"),
+        (".geojson", None, ["MultiPolygon", "Polygon"], "Unknown"),
+        (".geojson", True, ["MultiPolygon"], "MultiPolygon"),
+        (".geojson", False, ["MultiPolygon", "Polygon"], "Unknown"),
+    ],
+)
+def test_write_dataframe_promote_to_multi(
+    tmp_path,
+    naturalearth_lowres,
+    ext,
+    promote_to_multi,
+    expected_geometry_types,
+    expected_layer_geometry_type,
+):
+    input_gdf = read_dataframe(naturalearth_lowres)
+
+    output_path = tmp_path / f"test_promote{ext}"
+    write_dataframe(
+        input_gdf,
+        output_path,
+        promote_to_multi=promote_to_multi,
+    )
 
-    filename = tmp_path / "test.gpkg"
-    write_dataframe(df, filename, geometry_type="Unknown")
-    assert read_info(filename)["geometry_type"] == "Unknown"
+    assert output_path.exists()
+    output_gdf = read_dataframe(output_path)
+    geometry_types = sorted(output_gdf.geometry.type.unique())
+    assert geometry_types == expected_geometry_types
+    assert read_info(output_path)["geometry_type"] == expected_layer_geometry_type
 
-    write_dataframe(df, filename, geometry_type="Polygon")
-    assert read_info(filename)["geometry_type"] == "Polygon"
 
-    write_dataframe(df, filename, geometry_type="MultiPolygon")
-    assert read_info(filename)["geometry_type"] == "MultiPolygon"
+@pytest.mark.parametrize(
+    "ext, promote_to_multi, layer_geometry_type, "
+    "expected_geometry_types, expected_layer_geometry_type",
+    [
+        (".fgb", None, "Unknown", ["MultiPolygon"], "Unknown"),
+        (".geojson", False, "Unknown", ["MultiPolygon", "Polygon"], "Unknown"),
+        (".geojson", None, "Unknown", ["MultiPolygon", "Polygon"], "Unknown"),
+        (".geojson", None, "Polygon", ["MultiPolygon", "Polygon"], "Unknown"),
+        (".geojson", None, "MultiPolygon", ["MultiPolygon", "Polygon"], "Unknown"),
+        (".geojson", None, "Point", ["MultiPolygon", "Polygon"], "Unknown"),
+        (".geojson", True, "Unknown", ["MultiPolygon"], "MultiPolygon"),
+        (".gpkg", False, "Unknown", ["MultiPolygon", "Polygon"], "Unknown"),
+        (".gpkg", None, "Unknown", ["MultiPolygon"], "Unknown"),
```

Comment:
Should this line result in the same as the one above? (so if you explicitly set the type to Unknown, do not automatic promote_to_multi)

Reply:
I don't think so: 
- This would break the idea that the layer_geometry_type doesn't change/influence the geometry type of the data
- This would create a kind of odd interference between both parameters while it is already easy to accomplish this behaviour: if you just use promote_to_multi=False, this will evade promotion + will result in layer geometry type "Unknown". Check out the test test_write_dataframe_promote_to_multi where this is demonstrated.